lib/pbio/drv/usb/usb_stm32.c

```diff
@@ -164,11 +304,65 @@ PROCESS_THREAD(pbdrv_usb_process, ev, data) {
 
     PROCESS_BEGIN();
 
+    // Prepare to receive the first packet
+    USBD_Pybricks_ReceivePacket(&husbd);
+
+    etimer_set(&timer, 500);
+
     for (;;) {
         PROCESS_WAIT_EVENT();
 
         if (bcd_busy) {
             bcd_busy = PT_SCHEDULE(pbdrv_usb_stm32_bcd_detect(&bcd_pt));
+
+            if (bcd_busy) {
+                continue;
+            }
+        }
+
+        if (pbsys_status_test(PBIO_PYBRICKS_STATUS_SHUTDOWN_REQUEST) ||
+            pbsys_status_test(PBIO_PYBRICKS_STATUS_SHUTDOWN)) {
+            if (pbio_oneshot(true, &pwrdn_oneshot)) {
+                USBD_Stop(&husbd);
+            }
+
+            continue;
+        }
+
+        if (usb_in_sz) {
+            if ((usb_in_sz >= UUID_SZ) &&
+                pbio_uuid128_le_compare(usb_in_buf, pbio_pybricks_command_event_char_uuid)) {
```

Comment:
Do we actually need to include the UUID here? In other words, it seems like the endpoint number should be enough to indicate that anything received on this endpoint is a Pybricks command.

Reply:
See earlier comment about multiple endpoints. Assuming we stick with a single endpoint, it is good to check that the command UUID is present in case a second UUID is needed later.

Reply:
Changed to use messages.